goas/cl/ledger/src/constraint.rs

```diff
@@ -0,0 +1,87 @@
+use cl::Constraint;
+use ledger_proof_statements::constraint::ConstraintPublic;
+
+use crate::error::Result;
+
+#[derive(Debug, Clone)]
+pub struct ConstraintProof {
+    pub risc0_id: [u32; 8],
+    pub risc0_receipt: risc0_zkvm::Receipt,
+}
+
+pub fn risc0_constraint(risc0_id: [u32; 8]) -> Constraint {
+    // Commit to a RISC0 ID for use as a note constraint
+
+    let mut bytes = [0u8; 32];
+
+    for (i, word) in risc0_id.iter().enumerate() {
+        let word_bytes = word.to_le_bytes();
+        bytes[i * 4] = word_bytes[0];
+        bytes[i * 4 + 1] = word_bytes[1];
+        bytes[i * 4 + 2] = word_bytes[2];
+        bytes[i * 4 + 3] = word_bytes[3];
+    }
```

Comment:
nit: `risc0_id.into_iter().flat_map(|word| word.to_le_bytes()).try_into().unwrap()`?

Reply:
no `try_into()` for [u8;32] unfortunately